src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp

```diff
@@ -4632,6 +4632,69 @@ class StubGenerator: public StubCodeGenerator {
     return entry;
   }
 
+  // a1 = r1 - string1 address
+  // a2 = r3 - string2 address
+  // result = r0 - return value. Already contains "false"
+  // cnt1 = r4 - amount of elements left to check
+  address generate_long_string_equals() {
+    Register a1 = r1, a2 = r3, result = r0, cnt1 = r4;
+    Label NOT_EQUAL, EQUAL, LOOP, SMALL_LOOP, POST_LOOP;
+    int loopThreshold = 4 * wordSize;
+
+    assert_different_registers(a1, a2, result, cnt1);
+
+    __ align(CodeEntryAlignment);
+
+    StubCodeMark mark(this, "StubRoutines", "long_string_equals");
+
+    address entry = __ pc();
+    __ enter();
+
+    // reset cnt1
+    __ add(cnt1, cnt1, loopThreshold);
+
+    // Main 32 byte comparison loop.
+    __ bind(LOOP);
+      __ ld1(v0, v1, __ T2D, Address(__ post(a1, loopThreshold)));
```

Comment:
You need to reserve v0-3 as temporaries in the .ad file string_equals patterns otherwise we might be overwriting live values here.

Reply:
There's no point pursuing the use of vector registers any more.